jiant/utils/utils.py

```diff
@@ -361,15 +401,40 @@ def get_elmo_mixing_weights(text_field_embedder, task=None):
     return params
 
 
-def get_batch_size(batch, keyword="input"):
+def format_output(obj, cuda_devices):
+    """
+    Format output based on whether model is using DataParallel or not. 
+    DataParallel necessitates objects to be gathered into GPU:0 to have 
+    dimension 0. 
+    This function will be used for scalar outputs of model forwards 
+    such as loss and n_exs.
+    """
+    if isinstance(cuda_devices, list):
+        if isinstance(obj, torch.Tensor) is False:
+            obj = torch.tensor(obj).cuda()
+        return obj.unsqueeze(0)
+    else:
+        return obj
+
+
+def uses_cuda(cuda_devices):
```

Comment:
combine into a single clause ?

`if isinstance(cuda_devices, list) or (isinstance(cuda_devices, int) and cuda_devices >= 0): ...`

Or even a single line if you want to live with long lines lol

Reply:
yeah not about that long lines life